[JEST-EXPO] Error on pathResolve

## 🐛 Bug Report

### Summary of Issue (just a few sentences)

I am using Jest-expo 38 for my tests and since my upgrade from 37 to 38, I got this error below.

### Environment - output of `expo diagnostics` & the platform(s) you're targeting

```
$ expo diagnostics

  Expo CLI 3.21.12 environment info:
    System:
      OS: Linux 5.6 Manjaro Linux
      Shell: 5.8 - /usr/bin/zsh
    Binaries:
      Node: 14.3.0 - /usr/bin/node
      Yarn: 1.22.4 - /usr/bin/yarn
      npm: 6.14.5 - /usr/bin/npm
    npmPackages:
      expo: ^38.0.0 => 38.0.3
      react: 16.11.0 => 16.11.0
      react-dom: 16.11.0 => 16.11.0
      react-native: https://github.com/expo/react-native/archive/sdk-38.0.0.tar.gz => 0.62.2
      react-native-web: ~0.11.7 => 0.11.7
    npmGlobalPackages:
      expo-cli: 3.21.12
```

```
> @ test /home/me/code
> jest

internal/validators.js:121
    throw new ERR_INVALID_ARG_TYPE(name, 'string', value);
    ^

TypeError [ERR_INVALID_ARG_TYPE]: The "path" argument must be of type string. Received an instance of Object
    at validateString (internal/validators.js:121:11)
    at Object.resolve (path.js:980:7)
    at Object.<anonymous> (/home/me/code/node_modules/jest-expo/bin/jest.js:20:30)
    at Module._compile (internal/modules/cjs/loader.js:1200:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1220:10)
    at Module.load (internal/modules/cjs/loader.js:1049:32)
    at Function.Module._load (internal/modules/cjs/loader.js:937:14)
    at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:71:12)
    at internal/main/run_main_module.js:17:47 {
  code: 'ERR_INVALID_ARG_TYPE'
}
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! @ test: `jest`
npm ERR! Exit status 1
```

I fixed this issue by editing `node_modules/jest-expo/bin/jest.js` at line 20 by:
```
const jestProgramPath = path.resolve(jestPackagePath, jestPackageJson.bin.jest);
```

What do you think

--- a/packages/jest-expo/bin/jest.js
+++ b/packages/jest-expo/bin/jest.js
@@ -17,7 +17,10 @@ const process = require('process');
 // If you need to run Jest with the JS debugger enabled, run Jest directly. It is usually under
 // node_modules/jest/bin/jest.js.
 const jestPackagePath = path.dirname(require.resolve('jest/package.json'));
-const jestProgramPath = path.resolve(jestPackagePath, jestPackageJson.bin);
+const jestProgramPath = path.resolve(
+  jestPackagePath,
+  jestPackageJson.bin.jest || jestPackageJson.bin
+);
 const jestProgramArgs = process.argv.slice(2);
 const jestWithArgs = [jestProgramPath].concat(jestProgramArgs);
 const result = childProcess.spawnSync('node', jestWithArgs, { stdio: 'inherit' });

